### PR TITLE
DAOS-7444 control: Add support for dmg pool get-prop

### PR DESCRIPTION
**Pull request by mjmac**
Completely reworks dmg's support for pool properties:
  * Move name->number resolution client-side, to simplify
    server-side gRPC handlers and also to provide better
    CLI argument parsing/help/completion.
  * Pool create now accepts an optional set of properties
    so that a followup set-prop isn't necessary.
  * Setting props now has the same key:val format as the
    daos utility.

Signed-off-by: Michael MacDonald <mjmac.macdonald@intel.com>

### CODE REVIEW

File: src/mgmt/srv_drpc.c [hunk 1]
```diff
@@ -801,116 +870,189 @@ void ds_mgmt_drpc_pool_set_prop(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
...
+	d_rank_list_free(svc_ranks);
+out:
+	daos_prop_free(new_props);
+
+	resp.status = rc;
+	len = mgmt__pool_set_prop_resp__get_packed_size(&resp);
+	D_ALLOC(body, len);
+	if (body == NULL) {
+		drpc_resp->status = DRPC__STATUS__FAILED_MARSHAL;
+		D_ERROR("Failed to allocate drpc response body\n");
```
Comment (ashleypittman) [inline src/mgmt/srv_drpc.c:902]: I've removed a couple of these elsewhere but please don't log D_ALLOC() failures, the macro will do that for you.

Comment (mjmac) [reply src/mgmt/srv_drpc.c:902]: Ah, ok. Will remove those.

File: src/mgmt/tests/srv_drpc_tests.c [hunk 2]
```diff
@@ -925,26 +939,202 @@ test_drpc_pool_set_prop_success(void **state)
...
+	Drpc__Response		resp = DRPC__RESPONSE__INIT;
+	Mgmt__PoolGetPropReq	req = MGMT__POOL_GET_PROP_REQ__INIT;
+	int			prop_number = DAOS_PROP_PO_LABEL;
+	char			*prop_val = "foo";
+
+	ds_mgmt_pool_get_prop_out = daos_prop_alloc(1);
+	assert_non_null(ds_mgmt_pool_get_prop_out);
+	ds_mgmt_pool_get_prop_out->dpp_entries[0].dpe_type = prop_number;
+	D_STRNDUP(ds_mgmt_pool_get_prop_out->dpp_entries[0].dpe_str,
+		  prop_val, 4);
```
Comment (johannlombardi) [inline src/mgmt/tests/srv_drpc_tests.c:1123]: @mjmac sorry, i change the value above to 4 to fix the build issue on fedora, but it sounds like it broke the unit test.

**Comment by mjmac** [comment]
Force-pushed one last time to get the PR into shape for formal review. The unit tests build/pass locally.

**Review by johannlombardi — DISMISSED** [review]

**Comment by ashleypittman** [comment]
> Test stage NLT completed with status FAILURE. https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-6052/18/execution/node/349/log

This is an intermittent failure, I've added a suppression for it in https://github.com/daos-stack/daos/pull/6114

**Review by johannlombardi — APPROVED** [review]

**Comment by mjmac** [comment]
OK, this is finally looking stable enough to ask for reviews. It's a big PR, but there wasn't really a sensible way to break it up because it's all related to pool property handling. As I was looking at adding support for get-prop, I realized that the way set-prop had been done (by me) really didn't make much sense.

I have broken it into two commits to make it easier for reviewers to focus on their area of interest:

- [Control Plane Changes](https://github.com/daos-stack/daos/pull/6052/commits/684c9386edd8094e0c58c26e226c4bf9079dfee0)

- [Engine Changes](https://github.com/daos-stack/daos/pull/6052/commits/07d2106b0ec60a2ecaf6c230dbbfb04942888df8)

**Review by tanabarr — APPROVED** [review]
Go changes look good

**Pull request merged**